engine/src/main/java/org/terasology/world/block/BlockRegion.java

```diff
@@ -44,6 +44,7 @@
     public final AABBi aabb = new AABBi();
 
     public BlockRegion() {
+        this.setMin(0, 0, 0).setMax(0, 0, 0);
```

Comment:
This changes the semantics of `new BlockRegion()` from being some invalid infinite something (the empty region, so to say) to the non-empty BlockRegion containing exactly (0,0,0).

This allows to express `createFromCenterAndExtents` conveniently via
```java
new BlockRegion().translate(center).addExtents(extents)
```

Not sure whether the previous semantics was used somewhere, or whether we need the empty region to be a thing. We could add that to `BlockRegions`:

```java
public static BlockRegion empty() {
  return new BlockRegion(new AABBi());
}
```

Reply:
a static method to define an empty blockregion seems kind of strange. It makes more sense to me to have the initial state as undefined. `setMin(0, 0, 0).setMax(0, 0, 0)` would yield a region that is 1x1x1 units in size. min is 0,0,0 and max is 1,1,1

Reply:
hm, I probably don't get how these AABBs and regions work exactly :monocle_face: 

I thought that `setMin(0, 0, 0).setMax(0, 0, 0)` would create a region of size 0x0x0, containing a single point (0,0,0). This is basically a **point**.

With `setMin(0, 0, 0).setMax(0, 0, 1)` you'd create a region of size 0x0x1 containing 2 points. This is a **line**.

With `setMin(0, 0, 0).setMax(0, 1, 1)` you'd create a region of size 0x1x1 containing 4 points. This is a **plane** (surface).

With `setMin(0, 0, 0).setMax(1, 1, 1)` you'd create a unit cube of size 1x1x1 containing 8 points. This is a **volume**.

Is this understanding wrong?

Reply:
Another interpretation/visualization of `setMin(0, 0, 0).setMax(0, 0, 0)` would be to think of the region that contains exactly the block at position (0,0,0), i.e., the region going from min (-.5, -.5, -.5) to max (.5, .5, .5) and being of size 1x1x1.

Reply:
That explanation helps - I'll make sure to add that in the JavaDoc of BlockRegion for further reference.